### PR TITLE
[PW-2400]Add paypal method details

### PR DESCRIPTION
**Pull request by AlexandrosMor**
**Description**
Add paypal method details class

**Tested scenarios**

**Fixed issue**:  #259

### CODE REVIEW

File: Adyen/Model/Checkout/PayPalDetails.cs [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+using Newtonsoft.Json;
+using Newtonsoft.Json.Converters;
+
+namespace Adyen.Model.Checkout
+{
+    /// <summary>
+    /// 
+    /// </summary>
+    [DataContract]
+    public class PaypalDetails : IOneOfPaymentRequestPaymentMethod
```
Comment (Aleffio) [inline Adyen/Model/Checkout/PayPalDetails.cs:33]:
```suggestion
    public class PayPalDetails : IOneOfPaymentRequestPaymentMethod
```

File: Adyen/Model/Checkout/PayPalDetails.cs [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+        public string Type { get; set; } = "paypal";
+
+        /// <summary>
+        /// Get the string presentation of the object
+        /// </summary>
+        /// <returns>String presentation of the object</returns>
+        public override string ToString()
+        {
+            var sb = new StringBuilder();
+            sb.Append("class PaypalDetails {\n");
```
Comment (Aleffio) [inline Adyen/Model/Checkout/PayPalDetails.cs:74]:
```suggestion
            sb.Append("class PayPalDetails {\n");
```

File: Adyen/Model/Checkout/PayPalDetails.cs [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+
+            /// <summary>
+            /// Enum SDK for value: SDK
+            /// </summary>
+            [EnumMember(Value = "redirect")]
+            Redirect = 1,
+        }
+
+        /// <summary>
+        /// Paypal subtype
```
Comment (Aleffio) [inline Adyen/Model/Checkout/PayPalDetails.cs:52]:
```suggestion
        /// PayPal subtype
```

File: Adyen/Model/Checkout/PayPalDetails.cs [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+        [JsonProperty(PropertyName = "subtype")]
+        public SubtypeEnum Subtype;
+
+        /// <summary>
+        /// **paypal**
+        /// </summary>
+        /// <value>**paypal**</value>
+        [DataMember(Name = "type", EmitDefaultValue = false)]
+        [JsonProperty(PropertyName = "type")]
+        public string Type { get; set; } = "paypal";
```
Comment (rikterbeek) [inline Adyen/Model/Checkout/PayPalDetails.cs:65]: can we make from paypal an constant ? see how we do it in java: https://github.com/Adyen/adyen-java-api-library/blob/develop/src/main/java/com/adyen/model/checkout/details/PayPalDetails.java#L43

File: Adyen/Model/Checkout/PayPalDetails.cs [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+using Newtonsoft.Json;
+using Newtonsoft.Json.Converters;
+
+namespace Adyen.Model.Checkout
+{
+    /// <summary>
+    /// 
+    /// </summary>
+    [DataContract]
+    public class PayPalDetails : IOneOfPaymentRequestPaymentMethod
```
Comment (rikterbeek) [inline Adyen/Model/Checkout/PayPalDetails.cs:33]: IOneOfPaymentRequestPaymentMethod let's remove this name to IPaymentMethodDetails, let do this for all payment methods.

**Pull request merged**